src/group_theory/group_action/fixing_subgroup.lean

```diff
@@ -0,0 +1,260 @@
+/-
+Copyright (c) 2022 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+
+import group_theory.subgroup.basic
+import group_theory.group_action.basic
+import order.order_dual
+
+/-!
+
+# Fixing submonoid, fixing subgroup of an action
+
+In the presence of of an action of a monoid or a group,
+this file defines the fixing submonoid or the fixing subgroup,
+and relates it to the set of fixed points via a Galois connection.
+
+## Main definitions
+
+* `fixing_submonoid M s` :
+in the presence of `mul_action M α` (with `monoid M`)
+it is the `submonoid M` consisting of elements which fix `s : set α` pointwise.
+
+* `fixing_submonoid_fixed_points_connection M α` is the `galois_connection`
+that relates `fixing_submonoid` with `fixed_points`.
+
+* `fixing_submonoid_of_union` and `fixing_submonoid_of_Union` are consequences
+of the Galois connection, as well as `fixed_points_of_sup` and `fixed_points_of_supr`.
+
+* `fixing_subgroup M s` :
+in the presence of `mul_action M α` (with `group M`)
+it is the `subgroup M` consisting of elements which fix `s : set α` pointwise.
+
+* `fixing_subgroup_fixed_points_connection M α` is the `galois_connection`
+that relates `fixing_subgroup` with `fixed_points`.
+
+* `fixing_subgroup_of_union` and `fixing_subgroup_of_Union` are consequences
+of the Galois connection,
+as well as `fixed_points_of_group_of_sup` and `fixed_points_of_group_of_supr`.
+
+* The file starts with some lemmas that allow to rewrite `antitone` into `monotone`
+in various cases.
+Apparently the `monotone.dual` (and analogues) were not sufficient, so I wrote `monotone.dual_iff`
+that works in both directions.
+
+TODO :
+
+* Adjust names as needed
+
+* Decide what needs to be done with the antitonicity section. In particular, the
+proofs need two directions, but the proof term is exactly the same (up to types).
+
+* Maybe other lemmas are useful
+
+* Treat semigroups ?
+
+-/
+
+section antitonicity
```

Comment:
All the lemmas here would be good simp lemmas if started in the opposite direction